library/init/data/list/lemmas.lean

```diff
@@ -154,6 +158,24 @@ by induction l; simph
 @[simp] lemma map_reverse (f : α → β) (l : list α) : map f (reverse l) = reverse (map f l) :=
 by induction l; simph
 
+lemma reverse_core_append' (xs ys zs : list α) :
```

Comment:
I'm not sure why, I'd guess because it's mostly viewed as an internal function just needed to implement reverse.